src/webfaf/blueprints/celery_tasks.py

```diff
@@ -141,22 +143,74 @@ def add_argument(self, *args, **kwargs):
 
         self.F.argparse_fields[kwargs["dest"]] = kwargs
 
-    def add_opsys(self, multiple=False, required=False):
+    def add_argument_group(self, *args, **kwargs): # pylint: disable=unused-argument
+        return ActionFormArgGroup(self.F)
+
+    def add_mutually_exclusive_group(self, *args, **kwargs): # pylint: disable=unused-argument
+        return ActionFormArgGroup(self.F, mutually_exclusive=True)
+
+    def add_opsys(self, multiple=False, required=False, positional=False, with_rel=False, helpstr=None): # pylint: disable=unused-argument
         if required:
             vs = [validators.Required()]
         else:
             vs = [validators.Optional()]
 
+        choice_lst = [(osplugin.name, osplugin.nice_name) for osplugin in systems.values()]
+        arg_str = "opsys"
+
+        if positional:
+            arg_str = "OPSYS"
+
+            q = db.session.query(OpSys.name)
+            choice_lst = [(a[0], a[0]) for a in q]
+
+            if with_rel:
+                q = q.join(OpSysRelease)
+                q = q.with_entities(OpSys.name, OpSysRelease.version)
+                choice_lst = [(a[0] + " " + a[1], a[0] + " " + a[1]) for a in q]
```

Comment:
Can you factor this out? It seems a bit too busy around here. It would seem better to construct the list (or generator) of strings above and construct only the pair here.

Reply:
I made the DB lines into one and moved the DB queries out of the field constructor calls in the other methods that you commented on. I also tried construction the lists and the tuples in two steps like you suggested but it looked a little over the top to me.